microservices/services/contest-service/src/main/java/org/xcolab/service/contest/domain/contest/ContestDaoImpl.java

```diff
@@ -335,6 +347,74 @@ public boolean existsWithScheduleId(long contestScheduleId) {
 
     @Override
     public boolean delete(long contestPK) {
-        return dslContext.deleteFrom(CONTEST).where(CONTEST.CONTEST_PK.eq(contestPK)).execute() > 0;
+        final boolean[] result = new boolean[1];
+        dslContext.transaction(configuration -> {
+            DSLContext ctx = DSL.using(configuration);
+
+            deleteContestData(ctx, contestPK);
+            deleteContestPhases(ctx, contestPK);
+            deleteOrphanProposals(ctx);
+
+            result[0] = deleteContest(ctx, contestPK);
+        });
+        return result[0];
+    }
+
+    private static boolean deleteContest(DSLContext ctx, long contestPK) {
+        return ctx.deleteFrom(CONTEST)
+                .where(CONTEST.CONTEST_PK.eq(contestPK))
+                .execute() > 0;
+    }
+
+    private static void deleteContestData(DSLContext ctx, long contestPK) {
+        // Delete team members
+        ctx.deleteFrom(CONTEST_TEAM_MEMBER)
+                .where(CONTEST_TEAM_MEMBER.CONTEST_ID.eq(contestPK))
+                .execute();
+        // Delete contest translations
+        ctx.deleteFrom(CONTEST_TRANSLATION)
+                .where(CONTEST_TRANSLATION.CONTEST_ID.eq(contestPK))
+                .execute();
+        // Delete contest discussions
+        ctx.deleteFrom(CONTEST_DISCUSSION)
+                .where(CONTEST_DISCUSSION.CONTEST_ID.eq(contestPK))
+                .execute();
+        // Retrieve contest discussion thread id.
+        Long threadId = ctx.select(CONTEST.DISCUSSION_GROUP_ID)
+                .from(CONTEST)
+                .where(CONTEST.CONTEST_PK.eq(contestPK))
+                .fetchOne()
+                .value1();
```

Comment:
In another function you used `.fetchInto(Long.class)`; for consistency you could use `into(Long.class)` here.

Reply:
Done.